src/treesa.jl

```diff
@@ -156,82 +116,76 @@ Optimize the einsum contraction pattern using the simulated annealing on tensor
     end
+
     ###### Stage 1: preprocessing ######
     labels = _label_dict(ixs, iy)  # map labels to integers
     inverse_map = Dict([v=>k for (k,v) in labels])  # the inverse transformation, map integers to labels
     log2_sizes = [log2.(size_dict[inverse_map[i]]) for i=1:length(labels)]   # use `log2` sizes in computing time
     if ntrials <= 0  # no optimization at all, then 1). initialize an expression tree and 2). convert back to nested einsum.
         best_tree = _initializetree(code, size_dict, initializer; greedy_method=greedy_method)
-        return SlicedEinsum(LT[], NestedEinsum(best_tree, inverse_map))
+        return NestedEinsum(best_tree, inverse_map)
     end
+
     ###### Stage 2: computing ######
-    # create vectors to store optimized 1). expression tree, 2). time complexities, 3). space complexities, 4). read-write complexities and 5). slicing information.
-    trees, tcs, scs, rws, slicers = Vector{ExprTree}(undef, ntrials), zeros(ntrials), zeros(ntrials), zeros(ntrials), Vector{Slicer}(undef, ntrials)
+    # create vectors to store optimized 1). expression tree, 2). time complexities, 3). space complexities, 4). read-write complexities.
+    trees, tcs, scs, rws = Vector{ExprTree}(undef, ntrials), zeros(ntrials), zeros(ntrials), zeros(ntrials)
     @threads for t = 1:ntrials  # multi-threading on different trials, use `JULIA_NUM_THREADS=5 julia xxx.jl` for setting number of threads.
+
         # 1). random/greedy initialize a contraction tree.
         tree = _initializetree(code, size_dict, initializer; greedy_method=greedy_method)
-        # 2). optimize the `tree` and `slicer` in a inplace manner.
-        slicer = Slicer(log2_sizes, nslices, Int[labels[l] for l in fixed_slices])
-        optimize_tree_sa!(tree, log2_sizes, slicer; sc_target=sc_target, βs=βs, niters=niters, sc_weight=sc_weight, rw_weight=rw_weight)
+
+        # 2). optimize the `tree` in a inplace manner.
+        optimize_tree_sa!(tree, log2_sizes; sc_target=sc_target, βs=βs, niters=niters, sc_weight=sc_weight, rw_weight=rw_weight)
+
         # 3). evaluate time-space-readwrite complexities.
-        tc, sc, rw = tree_timespace_complexity(tree, slicer.log2_sizes)
+        tc, sc, rw = tree_timespace_complexity(tree, log2_sizes)
         @debug "trial $t, time complexity = $tc, space complexity = $sc, read-write complexity = $rw."
-        trees[t], tcs[t], scs[t], rws[t], slicers[t] = tree, tc, sc, rw, slicer
+        trees[t], tcs[t], scs[t], rws[t] = tree, tc, sc, rw
     end
+
     ###### Stage 3: postprocessing ######
     # compare and choose the best solution
-    best_tree, best_tc, best_sc, best_rw, best_slicer = first(trees), first(tcs), first(scs), first(rws), first(slicers)
+    best_tree, best_tc, best_sc, best_rw = first(trees), first(tcs), first(scs), first(rws)
     for i=2:ntrials
         if scs[i] < best_sc || (scs[i] == best_sc && exp2(tcs[i]) + rw_weight * exp2(rws[i]) < exp2(best_tc) + rw_weight * exp2(rws[i]))
```

Comment:
The comparison here is wrong and needs to be improved (a lagacy issue). I think we should stick to using the cost?